pkg/puppet/puppet.go

```diff
@@ -354,6 +354,13 @@ func (p *Puppet) writeHieraData(puppetPath string, cluster interfaces.Cluster) e
 			variables = append(variables, fmt.Sprintf(`kubernetes_addons::cluster_autoscaler::instance_pool_name: "%s"`, workerInstancePoolName))
 		}
 
+		if len(instancePool.Config().Labels) > 0 {
+			variables = append(variables, fmt.Sprintf("kubernetes::kubelet::node_labels:\n%s", instancePool.Labels()))
+		}
+		if len(instancePool.Config().Taints) > 0 {
+			variables = append(variables, fmt.Sprintf("kubernetes::kubelet::node_taints:\n%s", instancePool.Taints()))
+		}
```

Comment:
I think this will override node-taints and labels we currently set here:

https://github.com/jetstack/tarmak/blob/master/puppet/modules/kubernetes/manifests/kubelet.pp#L112

We need to make sure that master nodes are still label and taint like that and that worker nodes are actually labeld with role="worker".

We should only be able to override that by using the same key in the tarmak.yaml

This seems to be rather complex so ideally I want to test those cases.

Reply:
How about something like this? https://github.com/jetstack/tarmak/pull/369/commits/60511e45148c0f5f8320de30b867856e993c3933

Reply:
(drive-by review): how is variable/string escaping handled? What if I specify a taint with `:` in it or something?

Reply:
I've had a go at making this a little more robust & added some better error reporting in passing too. https://github.com/jetstack/tarmak/pull/369/commits/d150d9ed92aaf03f99803d08a8ab6c598bd83bce